src/string_util.cc

```diff
@@ -160,13 +160,19 @@ std::string StrFormat(const char* format, ...) {
   return tmp;
 }
 
-void ReplaceAll(std::string* str, const std::string& from,
-                const std::string& to) {
-  std::size_t start = 0;
-  while ((start = str->find(from, start)) != std::string::npos) {
-    str->replace(start, from.length(), to);
-    start += to.length();
-  }
+std::string StrEscape(const std::string & s) {
+  std::string tmp;
```

Comment:
It's a review comment, it is good to address them, so that there is forward progress on proposed changes.
Not sure how these github suggestions work:
```suggestion
  std::string tmp;
  tmp.reserve(s.size());
```

Reply:
you do realize that std::string is already designed for this kind of concatenation? this is trading 0-1-2 reallocs (depending on the length of the key) with a guaranteed two calls (strlen,realloc), not sure if there's a real win here, unless it's an issue of coding style?

Reply:
We *know* the escaped string will be at least not shorter than the input string.
clang/gcc will not magically insert that pre-allocation.
Therefore, if we don't do it explicitly, we are relying on the sane behavior
of the `std::string` implementation in the C++ std library that is used.
* We will likely always have some allocs if the string after escaping does not fit within small-size-optimization.
* Without pre-allocation, if the source string does not fit within small-size-optimization,
and there won't be extra escape symbols, we will have allocations.
* WITH pre-allocation, if the source string does not fit within small-size-optimization,
and there won't be extra escape symbols, we will *NOT* have allocations. WIN.

It is well-known that `reserve()` is good, and it is quite obvious it won't hurt here.
I do not understand your point.